src/main/java/com/nike/cerberus/auth/connector/okta/OktaAuthHelper.java

```diff
@@ -154,12 +171,130 @@ protected AuthResult authenticateUser(final String username, final String passwo
+        if (MFA_FACTOR_NAMES.containsKey(factorProvider)) {
+            return MFA_FACTOR_NAMES.get(factorProvider);
+        }
+
+        return WordUtils.capitalizeFully(factorProvider);
+    }
+
+    /**
+     * Get list of enrolled MFA factors for a user
+     * @param userId  Okta user ID
+     * @return List of factors
+     */
+    protected List<Factor> getFactorsByUserId(final String userId) {
+
+        Preconditions.checkArgument(userId != null, "user id cannot be null.");
+
+        try {
+            return this.factorsApiClient.getUserLifecycleFactors(userId);
+        } catch (IOException e) {
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.INTERNAL_SERVER_ERROR)
+                    .withExceptionCause(e)
+                    .withExceptionMessage("Error parsing the embedded auth data from Okta.")
+                    .build();
+        }
+    }
+
+    /**
+     * Get list of enrolled MFA factors for a user from their authentication result
+     * @param authResult  Okta response object after successful authentication
+     * @return List of factors
+     */
+    protected List<Factor> getUserFactorsFromAuthResult(final AuthResult authResult) {
+
+        final EmbeddedAuthResponseDataV1 embeddedAuthData = this.getEmbeddedAuthData(authResult);
+
+        if (embeddedAuthData != null && embeddedAuthData.getFactors() != null) {
+            return embeddedAuthData.getFactors();
+        } else {
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.INTERNAL_SERVER_ERROR)
+                    .withExceptionMessage("Could not parse user factors from Okta response.")
+                    .build();
+        }
+    }
+
+    /**
+     * Get user id from authentication result
+     * @param authResult Okta response object after successful authentication
+     * @return The u
+     * ser ID
+     */
+    protected String getUserIdFromAuthResult(final AuthResult authResult) {
+        final EmbeddedAuthResponseDataV1 embeddedAuthData = this.getEmbeddedAuthData(authResult);
+
+        if (embeddedAuthData.getUser() != null) {
+            return embeddedAuthData.getUser().getId();
+        } else {
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.INTERNAL_SERVER_ERROR)
+                    .withExceptionMessage("Could not parse user data from Okta response.")
+                    .build();
+        }
+    }
+
+    /**
+     * Get username from authentication result
+     * @param authResult Okta response object after successful authentication
+     * @return The username
+     */
+    String getUserLoginFromAuthResult(final AuthResult authResult) {
+        final EmbeddedAuthResponseDataV1 embeddedAuthData = this.getEmbeddedAuthData(authResult);
+
+        try {
+            return embeddedAuthData.getUser().getProfile().getLogin();
+        } catch(NullPointerException npe) {
```

Comment:
Catching NPE here seems slightly off to me.  Is there a more specific check for "Could not parse user data from Okta response"?  NPE is so general, it might mask other bugs.